// @nbit/utils@1.0.1-beta.7/es/decimal/index.js
import Decimal from 'decimal.js';
import { isString } from 'lodash';
/**
 * 设置精度位 - 最大有效字位数
 * @param val
 */

function setPrecision(val) {
  Decimal.set({
    precision: val
  });
}
/**
 * 将安全金额的字符串还原回原始数字字符串，方便和后续的 decimal 计算
 * @param safeCurrency 安全金额的字符串
 * @returns 去除掉其他符号的数字字符串
 */


function getSafeCurrency(safeCurrency) {
  if (!safeCurrency) {
    return '0';
  }

  if (isString(safeCurrency)) {
    if (safeCurrency === '--') {
      return '0';
    }

    var number = new Decimal(safeCurrency.trim());

    if (isNaN(number.toNumber())) {
      return '0';
    }

    return number.toString();
  }

  return safeCurrency;
}
/** 将值安全转换为 Decimal 数据 */


function getSafeDecimal(value) {
  if (value instanceof Decimal) {
    return value;
  }

  if (!value) {
    return new Decimal(0);
  }

  if (typeof value === 'string') {
    return new Decimal(getSafeCurrency(value));
  }

  return new Decimal(value);
}
/**
 * 添加安全的计算工具
 */


var SafeCalcUtil = {
  mul: function mul(a, b) {
    return getSafeDecimal(a).mul(getSafeDecimal(b));
  },
  div: function div(a, b) {
    return getSafeDecimal(a).div(getSafeDecimal(b));
  },
  add: function add(a, b) {
    return getSafeDecimal(a).add(getSafeDecimal(b));
  },
  sub: function sub(a, b) {
    return getSafeDecimal(a).sub(getSafeDecimal(b));
  }
};
export { SafeCalcUtil, getSafeCurrency, getSafeDecimal, setPrecision };
