// haoma@2.31.0/lib/cli.js
#!/usr/bin/env node
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const deepmerge_1 = tslib_1.__importDefault(require("deepmerge"));
const globby_1 = tslib_1.__importDefault(require("globby"));
const json5_1 = tslib_1.__importDefault(require("json5"));
const rimraf_1 = tslib_1.__importDefault(require("rimraf"));
const cross_spawn_1 = tslib_1.__importDefault(require("cross-spawn"));
const yargs_1 = tslib_1.__importDefault(require("yargs"));
const path_1 = require("path");
const bundle_1 = require("./bundle");
const vtils_1 = require("vtils");
const compile_1 = require("./compile");
const fs_extra_1 = require("fs-extra");
const getBabelConfig_1 = require("./getBabelConfig");
const run_1 = require("./run");
require('@babel/register')({
    ...getBabelConfig_1.getBabelConfig({
        typescript: true,
        target: 'node',
    }),
    extensions: ['.js', '.ts'],
});
yargs_1.default
    .usage('Usage: $0 <command> [options]')
    .help('help', 'Show help')
    .alias('help', 'h')
    // Show help if no args
    // ref: https://github.com/yargs/yargs/issues/895
    .demandCommand(1, '')
    .command('init', 'Initialize the config files', yargs => {
    yargs
        .option('override', {
        alias: 'o',
        type: 'boolean',
        describe: 'Override existing files',
        default: false,
    })
        .option('jest', {
        alias: 'j',
        type: 'boolean',
        describe: 'Install jest',
        default: false,
    })
        .option('license', {
        alias: 'l',
        type: 'boolean',
        describe: 'Generate LICENSE',
        default: false,
    });
}, argv => {
    var _a, _b;
    const cwd = process.cwd();
    const { name: packageName, version: packageVersion, } = require('../package.json');
    // ========= 文件写入 =========
    {
        console.log('Write config files...');
        const writeable = (path, writer) => {
            const existing = fs_extra_1.existsSync(path);
            if (!existing || (existing && argv.override)) {
                writer(path);
                console.log(`✔️ Write ${path}`);
            }
        };
        // .gitignore
        const gitignoreFile = path_1.join(cwd, '.gitignore');
        const gitignoreContent = fs_extra_1.existsSync(gitignoreFile)
            ? fs_extra_1.readFileSync(gitignoreFile, 'utf8')
            : '';
        const gitignoreItems = gitignoreContent.split(/[\r\n]+/g);
        const newGitignoreItems = vtils_1.dedent `
          .DS_Store
          Thumbs.db
          node_modules
          coverage
          lib
          dist
          *.tsbuildinfo
          *.log*
        `.split(/[\r\n]+/g);
        const finalGitignoreItems = vtils_1.uniq([
            ...gitignoreItems,
            ...newGitignoreItems,
        ]);
        const finalGitignoreContent = finalGitignoreItems.join('\n');
        fs_extra_1.writeFileSync(gitignoreFile, finalGitignoreContent);
        console.log(`✔️ Write ${gitignoreFile}`);
        // .gitattributes
        writeable(path_1.join(cwd, '.gitattributes'), path => {
            fs_extra_1.writeFileSync(path, `${vtils_1.dedent `
              * text eol=lf

              *.png binary
              *.jpg binary
              *.gif binary
              *.jpeg binary
              *.mp3 binary
              *.aac binary
              *.mp4 binary
              *.sketch binary
              *.json linguist-language=JSON-with-Comments
            `}\n`);
        });
        // .editorconfig
        writeable(path_1.join(cwd, '.editorconfig'), path => {
            fs_extra_1.writeFileSync(path, `${vtils_1.dedent `
              # http://editorconfig.org
              root = true

              [*]
              indent_style = space
              indent_size = 2
              end_of_line = lf
              charset = utf-8
              trim_trailing_whitespace = true
              insert_final_newline = true

              # The JSON files contain newlines inconsistently
              [*.json]
              insert_final_newline = ignore

              # Minified JavaScript files shouldn't be changed
              [**.min.js]
              indent_style = ignore
              insert_final_newline = ignore

              [*.md]
              trim_trailing_whitespace = false
            `}\n`);
        });
        // .eslintrc.js
        writeable(path_1.join(cwd, '.eslintrc.js'), path => {
            fs_extra_1.writeFileSync(path, `${vtils_1.dedent `
              /** @type import('${packageName}').ESLintConfig */
              module.exports = require('${packageName}').getESLintConfig()
            `}\n`);
        });
        // .eslintignore
        writeable(path_1.join(cwd, '.eslintignore'), path => {
            fs_extra_1.writeFileSync(path, `${vtils_1.dedent `
              !**/.*
              node_modules
              lib
              dist
            `}\n`);
        });
        // .prettierrc.js
        writeable(path_1.join(cwd, '.prettierrc.js'), path => {
            fs_extra_1.writeFileSync(path, `${vtils_1.dedent `
              /** @type import('${packageName}').PrettierConfig */
              module.exports = require('${packageName}').getPrettierConfig()
            `}\n`);
        });
        // .prettierignore
        writeable(path_1.join(cwd, '.prettierignore'), path => {
            fs_extra_1.writeFileSync(path, `${vtils_1.dedent `
              node_modules
              package-lock.json
              yarn.lock
              pnpm-lock.yaml
              lib
              dist
            `}\n`);
        });
        // tsconfig.json
        writeable(path_1.join(cwd, 'tsconfig.json'), path => {
            fs_extra_1.writeFileSync(path, JSON.stringify({
                compilerOptions: {
                    strict: true,
                    target: 'ESNext',
                    module: 'ESNext',
                    moduleResolution: 'node',
                    declaration: true,
                    removeComments: false,
                    esModuleInterop: true,
                    allowSyntheticDefaultImports: true,
                    experimentalDecorators: true,
                    skipLibCheck: true,
                    sourceMap: false,
                    preserveSymlinks: true,
                    newLine: 'LF',
                    jsx: 'react',
                    lib: ['esnext', 'dom'],
                },
            }, null, 2));
        });
        if (argv.license) {
            const gitUser = cross_spawn_1.default
                .sync('git', ['config', '--get', 'user.name'])
                .output.join('')
                .trim() || 'AUTHOR_NAME';
            const gitEmail = cross_spawn_1.default
                .sync('git', ['config', '--get', 'user.email'])
                .output.join('')
                .trim() || 'AUTHOR_EMAIL';
            // LICENSE
            writeable(path_1.join(cwd, 'LICENSE'), path => {
                fs_extra_1.writeFileSync(path, `${vtils_1.dedent `
                MIT License

                Copyright (c) ${String(new Date().getFullYear())}-present ${gitUser} <${gitEmail}>

                Permission is hereby granted, free of charge, to any person obtaining a copy
                of this software and associated documentation files (the "Software"), to deal
                in the Software without restriction, including without limitation the rights
                to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
                copies of the Software, and to permit persons to whom the Software is
                furnished to do so, subject to the following conditions:

                The above copyright notice and this permission notice shall be included in all
                copies or substantial portions of the Software.

                THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
                IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
                FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
                AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
                LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
                OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
                SOFTWARE.
            `}\n`);
            });
        }
        if (argv.jest) {
            writeable(path_1.join(cwd, 'jest.config.js'), path => {
                fs_extra_1.writeFileSync(path, `${vtils_1.dedent `
                /** @type import('${packageName}').JestConfig */
                module.exports = require('${packageName}').getJestConfig()
              `}\n`);
            });
        }
        console.log('✔️ Write config files');
    }
    // ========= 依赖安装 =========
    {
        console.log('Install dependencies...');
        const currentPackageJson = path_1.join(cwd, 'package.json');
        const currentTsConfig = path_1.join(cwd, 'tsconfig.json');
        const currentPackageInfo = fs_extra_1.existsSync(currentPackageJson)
            ? JSON.parse(fs_extra_1.readFileSync(currentPackageJson).toString())
            : {};
        const currentPackageDeps = new Set(Object.keys({
            ...currentPackageInfo.dependencies,
            ...currentPackageInfo.devDependencies,
        }));
        const needInstallPackages = [
            {
                name: packageName,
                version: `^${packageVersion.split('.')[0]}`,
            },
            { name: 'typescript', version: '^4' },
            { name: 'eslint', version: '^7' },
            { name: 'prettier', version: '^2' },
            { name: 'husky', version: '^4' },
            { name: 'lint-staged', version: '^10' },
            { name: 'standard-version', version: '^9' },
            ...(!argv.jest
                ? []
                : [
                    { name: 'jest', version: '^26' },
                    { name: 'codecov', version: '^3' },
                    { name: 'typescript-snapshots-plugin', version: '^1' },
                ]),
        ].filter(pkg => !currentPackageDeps.has(pkg.name));
        const userAgent = process.env.npm_config_user_agent;
        const userAgentX = path_1.basename(process.env._ || '');
        const usePackageManager = (userAgentX &&
            (userAgentX === 'yarn'
                ? 'yarn'
                : userAgentX === 'pnpx'
                    ? 'pnpm'
                    : userAgentX === 'npx'
                        ? 'npm'
                        : '')) ||
            (userAgent &&
                (userAgent.startsWith('yarn')
                    ? 'yarn'
                    : userAgent.startsWith('pnpm')
                        ? 'pnpm'
                        : userAgent.startsWith('npm')
                            ? 'npm'
                            : '')) ||
            (fs_extra_1.existsSync(path_1.join(cwd, 'yarn.lock'))
                ? 'yarn'
                : fs_extra_1.existsSync(path_1.join(cwd, 'pnpm-lock.yaml'))
                    ? 'pnpm'
                    : fs_extra_1.existsSync(path_1.join(cwd, 'package-lock.json'))
                        ? 'npm'
                        : '') ||
            'npm';
        if (usePackageManager === 'pnpm') {
            const currentNpmrc = path_1.join(cwd, '.npmrc');
            const currentNpmrcInfo = fs_extra_1.existsSync(currentNpmrc)
                ? fs_extra_1.readFileSync(currentNpmrc).toString()
                : '';
            if (!currentNpmrcInfo.includes('shamefully-hoist')) {
                const currentNodeModules = path_1.join(cwd, 'node_modules');
                rimraf_1.default.sync(currentNodeModules);
                fs_extra_1.writeFileSync(currentNpmrc, currentNpmrcInfo
                    .split('\n')
                    .filter(Boolean)
                    .concat('shamefully-hoist=true', '')
                    .join('\n'));
            }
        }
        if (needInstallPackages.length > 0) {
            cross_spawn_1.default.sync(usePackageManager, [
                'add',
                ...needInstallPackages.map(pkg => `${pkg.name}@${pkg.version}`),
                '-D',
            ], {
                stdio: 'inherit',
                cwd: cwd,
                env: process.env,
            });
        }
        const nextPackageInfo = JSON.parse(fs_extra_1.readFileSync(currentPackageJson).toString());
        if (!nextPackageInfo.husky && !nextPackageInfo['lint-staged']) {
            nextPackageInfo.husky = {
                hooks: {
                    'pre-commit': 'lint-staged',
                },
            };
            nextPackageInfo['lint-staged'] = {
                '*.{css,less,scss,sass,html,htm,vue,yml,yaml,json,md}': [
                    'prettier --write',
                ],
                '*.{js,jsx,ts,tsx}': ['eslint --fix', 'prettier --write'],
            };
        }
        const nextTsInfo = fs_extra_1.existsSync(currentTsConfig)
            ? json5_1.default.parse(fs_extra_1.readFileSync(currentTsConfig).toString())
            : {};
        if (!((_b = (_a = nextTsInfo.compilerOptions) === null || _a === void 0 ? void 0 : _a.plugins) === null || _b === void 0 ? void 0 : _b.some(plugin => plugin.name === 'typescript-snapshots-plugin'))) {
            nextTsInfo.compilerOptions = nextTsInfo.compilerOptions || {};
            nextTsInfo.compilerOptions.plugins =
                nextTsInfo.compilerOptions.plugins || [];
            nextTsInfo.compilerOptions.plugins.push({
                name: 'typescript-snapshots-plugin',
                snapshotCallIdentifiers: [
                    'toMatchSnapshot',
                    'toThrowErrorMatchingSnapshot',
                    'toMatchDiffSnapshot',
                ],
            });
        }
        fs_extra_1.writeFileSync(currentPackageJson, JSON.stringify(nextPackageInfo, null, 2));
        fs_extra_1.writeFileSync(currentTsConfig, JSON.stringify(nextTsInfo, null, 2));
        cross_spawn_1.default.sync('npx', ['prettier', '--write', currentPackageJson, currentTsConfig], {
            stdio: 'inherit',
            cwd: cwd,
            env: process.env,
        });
        console.log('✔️ Install dependencies');
    }
    // patch eslint 7 for vscode
    // issue: https://github.com/microsoft/vscode-eslint/issues/972
    {
        const vscodePath = path_1.join(cwd, '.vscode');
        const vscodeSettingsFile = path_1.join(vscodePath, 'settings.json');
        if (!fs_extra_1.existsSync(vscodePath)) {
            fs_extra_1.mkdirSync(vscodePath);
        }
        fs_extra_1.writeFileSync(vscodeSettingsFile, JSON.stringify(deepmerge_1.default(fs_extra_1.existsSync(vscodeSettingsFile)
            ? JSON.parse(fs_extra_1.readFileSync(vscodeSettingsFile, 'utf8'))
            : {}, {
            'eslint.options': {
                resolvePluginsRelativeTo: '.',
            },
        }), null, 2));
    }
})
    .command('run', 'Run a js/ts script with the babel register', () => undefined, async () => {
    await run_1.run({
        transformer: 'babel',
        args: process.argv.slice(3),
        cwd: process.cwd(),
        env: process.env,
    });
})
    .command('runs', 'Run a js/ts script with the swc register', () => undefined, async () => {
    await run_1.run({
        transformer: 'swc',
        args: process.argv.slice(3),
        cwd: process.cwd(),
        env: process.env,
    });
})
    .command('rune', 'Run a js/ts script with the esbuild register', () => undefined, async () => {
    await run_1.run({
        transformer: 'esbuild',
        args: process.argv.slice(3),
        cwd: process.cwd(),
        env: process.env,
    });
})
    .command('compile', 'Compile files', () => undefined, async () => {
    const configTsFile = path_1.join(process.cwd(), 'haoma-compile.config.ts');
    const configJsFile = path_1.join(process.cwd(), 'haoma-compile.config.js');
    const configFile = (await fs_extra_1.pathExists(configTsFile))
        ? configTsFile
        : configJsFile;
    if (!(await fs_extra_1.pathExists(configFile))) {
        throw new Error('找不到配置文件');
    }
    const config = vtils_1.castArray((require(configFile).default ||
        require(configFile)));
    for (const configItem of config) {
        const inputFiles = await globby_1.default(configItem.inputFiles, {
            cwd: process.cwd(),
            onlyFiles: true,
            absolute: true,
        });
        const outDir = path_1.resolve(process.cwd(), configItem.outDir);
        await compile_1.compile({
            ...configItem,
            inputFiles,
            outDir,
        });
    }
})
    .command('bundle [input]', 'Bundle a nodejs file using esbuild', yargs => {
    yargs
        .option('output', {
        alias: 'o',
        type: 'string',
        describe: 'Output file',
        demandOption: true,
    })
        .option('format', {
        alias: 'f',
        type: 'string',
        describe: 'Format',
        default: undefined,
    })
        .option('externals', {
        alias: 'e',
        type: 'array',
        describe: 'External packages',
        coerce: (value) => value.map(item => item.split(',')).flat(),
        default: [],
    })
        .option('minify', {
        alias: 'm',
        type: 'boolean',
        describe: 'Minify',
        default: false,
    })
        .option('nodeEnv', {
        type: 'string',
        describe: 'process.env.NODE_ENV',
        default: undefined,
    });
}, async (args) => {
    await bundle_1.bundle({
        input: path_1.resolve(process.cwd(), args.input),
        output: path_1.resolve(process.cwd(), args.output),
        format: args.format,
        externals: args.externals,
        minify: args.minify,
        nodeEnv: args.nodeEnv,
    });
}).argv;
