// haoma@2.31.0/lib/getPrettierConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPrettierConfig = void 0;
const tslib_1 = require("tslib");
const deepmerge_1 = tslib_1.__importDefault(require("deepmerge"));
const path_1 = require("path");
function getPrettierConfig(customConfig = {}, projectRoot = process.cwd()) {
    const paths = [
        path_1.join(__dirname, '../node_modules'),
        path_1.join(projectRoot, 'node_modules'),
    ];
    return deepmerge_1.default({
        printWidth: 80,
        tabWidth: 2,
        useTabs: false,
        semi: false,
        singleQuote: true,
        quoteProps: 'consistent',
        jsxSingleQuote: true,
        trailingComma: 'all',
        bracketSpacing: true,
        jsxBracketSameLine: true,
        arrowParens: 'avoid',
        endOfLine: 'lf',
        vueIndentScriptAndStyle: true,
        plugins: [
            require.resolve('prettier-plugin-packagejson', { paths }),
            require.resolve('prettier-plugin-sh', { paths }),
        ],
        overrides: [
            {
                files: ['.npmrc', '.yarnrc'],
                options: {
                    // @ts-ignore
                    parser: 'sh',
                },
            },
        ],
    }, customConfig);
}
exports.getPrettierConfig = getPrettierConfig;
