// haoma@2.31.0/lib/rollupDts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const workerpool_1 = tslib_1.__importDefault(require("workerpool"));
const api_extractor_1 = require("@microsoft/api-extractor");
const path_1 = require("path");
const vtils_1 = require("vtils");
function rollupDts(dtsFiles, includedPackages = []) {
    const consoleLog = console.log;
    console.log = vtils_1.noop;
    dtsFiles.forEach(dtsFile => {
        const config = api_extractor_1.ExtractorConfig.prepare({
            configObjectFullPath: path_1.join(process.cwd(), './api-extractor.json'),
            configObject: {
                projectFolder: process.cwd(),
                mainEntryPointFilePath: dtsFile,
                bundledPackages: includedPackages,
                apiReport: { enabled: false, reportFileName: 'report.api.md' },
                dtsRollup: { enabled: true, untrimmedFilePath: dtsFile },
                tsdocMetadata: { enabled: false },
                docModel: { enabled: false },
                compiler: {
                    tsconfigFilePath: path_1.join(process.cwd(), './tsconfig.json'),
                },
                newlineKind: 'lf',
            },
            packageJsonFullPath: path_1.join(process.cwd(), './package.json'),
            packageJson: {
                name: 'hello',
            },
        });
        api_extractor_1.Extractor.invoke(config, {
            localBuild: true,
        });
    });
    console.log = consoleLog;
}
workerpool_1.default.worker({
    rollupDts,
});
