// @nbit/chart-utils@1.0.0-beta.35/lib/const/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const timeLocaleLanguageMap = {
    'zh-CN': {
        '基本版': '基本版',
        '深度图': '深度图',
        '更多': '更多',
        '1m': '1分',
        '1time': '分时',
        '3m': '3分',
        '5m': '5分',
        '15m': '15分',
        '30m': '30分',
        '1h': '1小时',
        '2h': '2小时',
        '4h': '4小时',
        '6h': '6小时',
        '8h': '8小时',
        '12h': '12小时',
        '1D': '1日',
        '3D': '3日',
        '5D': '5日',
        '1W': '1周',
        '1M': '1月',
        '3M': '3月',
        '6M': '6月',
        '1Y': '1年',
        '图表设置': '图表设置',
        '指标设置': '指标设置',
        '时间周期': '时间周期',
        '图表高度': '图表高度',
        '偏低': '偏低',
        '默认': '默认',
        '偏高': '偏高',
        '取消': '取消',
        '展示周期': '展示周期',
        '重置': '重置',
        '确认': '确认',
        '必须选择五个时间': '必须选择五个时间',
        '主图': '主图',
        '副图': '副图',
        '时间': '时间',
        '开': '开',
        '高': '高',
        '低': '低',
        '收': '收',
        '涨跌幅': '涨跌幅',
        '振幅': '振幅',
        '量': '量',
        '额': '额',
        '成交': '成交',
        '笔': '笔',
        '均价': '均价',
        '买入': '买入',
        '查看明细': '查看明细',
        '卖出': '卖出',
        '数量': '数量',
        '价格': '价格',
        '手续费': '手续费',
        '订单详情': '订单详情',
        "价差幅度": "Spread",
        "委托价格": "Commission price",
        "累计挂单": "Accumulated pending orders",
    },
    'en-US': {
        '基本版': 'Original',
        '深度图': 'Depth',
        '更多': 'more',
        '1m': '1m',
        '1time': 'Time',
        '3m': '3m',
        '5m': '5m',
        '15m': '15m',
        '30m': '30m',
        '1h': '1h',
        '2h': '2h',
        '4h': '4h',
        '6h': '6h',
        '8h': '8h',
        '12h': '12h',
        '1D': '1D',
        '3D': '3D',
        '5D': '5D',
        '1W': '1W',
        '1M': '1M',
        '3M': '3M',
        '6M': '6M',
        '1Y': '1Y',
        '图表设置': 'Chart Settings',
        '指标设置': 'Indicator Setting',
        '时间周期': 'Cycle time',
        '图表高度': 'Chart Height',
        '偏低': 'Low',
        '默认': 'Default',
        '偏高': 'High',
        '取消': 'Cancel',
        '展示周期': 'Exhibition cycle',
        '重置': 'Reset',
        '确认': 'Confirm',
        '必须选择五个时间': 'Five times must be selected',
        '主图': 'Main drawing',
        '副图': 'Secondary drawing',
        '时间': 'Time',
        '开': 'Open',
        '高': 'High',
        '低': 'Low',
        '收': 'Close',
        '涨跌幅': 'Chg',
        '振幅': 'Amplitude',
        '量': 'Volume',
        '额': 'QuoteVolume',
        '成交': 'Deal',
        '笔': 'Pen',
        '均价': 'Average price',
        '买入': 'Buy',
        '查看明细': 'View details',
        '卖出': 'Sell',
        '数量': 'Count',
        '价格': 'Price',
        '手续费': 'Service charge',
        '订单详情': 'Order details',
        "价差幅度": "价差幅度",
        "委托价格": "价差幅度",
        "累计挂单": "累计挂单",
    },
};
const timeMap = {
    '1time': '1m',
    '1m': '1m',
    '3m': '3m',
    '5m': '5m',
    '15m': '15m',
    '30m': '30m',
    '1h': '1h',
    '2h': '2h',
    '4h': '4h',
    '6h': '6h',
    '8h': '8h',
    '12h': '12h',
    '1D': '1d',
    '3D': '3d',
    '5D': '5d',
    '1W': '1w',
    '1M': '1M',
    '3M': '3M',
    '6M': '6M',
    '1Y': '1y',
};
const tradingviewTimeMap = {
    '1time': '1',
    '1m': '1',
    '3m': '3',
    '5m': '5',
    '15m': '15',
    '30m': '30',
    '1h': '60',
    '2h': '120',
    '4h': '240',
    '6h': '360',
    '8h': '480',
    '12h': '720',
    '1D': '1440',
    '3D': '4320',
    '5D': '7200',
    '1W': '10080',
    '1M': '43200',
    '3M': '129600',
    '6M': '259200',
    '1Y': '518400',
};
exports.TradingviewIndicatorType = void 0;
(function (TradingviewIndicatorType) {
    TradingviewIndicatorType["ChartProperties"] = "chartProperties";
    TradingviewIndicatorType["InsertIndicator"] = "insertIndicator";
})(exports.TradingviewIndicatorType || (exports.TradingviewIndicatorType = {}));
const shareTimeList = [
    {
        unit: 'm',
        value: 1,
    },
    {
        unit: 'm',
        value: 5,
    },
    {
        unit: 'm',
        value: 15,
    },
    {
        unit: 'm',
        value: 30,
    },
    {
        unit: 'h',
        value: 1,
    },
    {
        unit: 'h',
        value: 2,
    },
    {
        unit: 'h',
        value: 4,
    },
    {
        unit: 'D',
        value: 1,
    },
    {
        unit: 'W',
        value: 1,
    },
    {
        unit: 'M',
        value: 1,
    },
];
const LanguageMapToChartMap = {
    'zh-CN': 'zh',
    'en-US': 'en',
    'ja-JP': 'ja',
    'ko-KR': 'ko',
    'th-TH': 'th_TH',
    'zh-TW': 'zh_TW',
    'fr-FR': 'fr',
    'de-DE': 'de_DE',
    'it-IT': 'it',
    'es-ES': 'es',
    'vi-VN': 'vi',
};
const ThemeChartMap = {
    light: 'Light',
    dark: 'Dark',
};
exports.WSThrottleTypeEnum = void 0;
(function (WSThrottleTypeEnum) {
    // 不需要延迟
    WSThrottleTypeEnum[WSThrottleTypeEnum["none"] = 0] = "none";
    // 覆盖式更新数据
    WSThrottleTypeEnum[WSThrottleTypeEnum["cover"] = 1] = "cover";
    // 增量更新数据
    WSThrottleTypeEnum[WSThrottleTypeEnum["increment"] = 2] = "increment";
})(exports.WSThrottleTypeEnum || (exports.WSThrottleTypeEnum = {}));
exports.ThemeEnum = void 0;
(function (ThemeEnum) {
    ThemeEnum["light"] = "light";
    ThemeEnum["dark"] = "dark";
})(exports.ThemeEnum || (exports.ThemeEnum = {}));
const bollKList = [
    {
        name: 'UP',
        value: 'upper',
    },
    {
        name: 'MB',
        value: 'mid',
    },
    {
        name: 'DN',
        value: 'lower',
    },
];
const macdKList = [
    {
        name: 'dea',
        value: 'dea',
    },
    {
        name: 'dif',
        value: 'dif',
    },
    {
        name: 'macd',
        value: 'macd',
    },
];
const kdjKList = [
    {
        name: 'K',
        value: 'k',
    },
    {
        name: 'D',
        value: 'd',
    },
    {
        name: 'J',
        value: 'j',
    },
];

exports.LanguageMapToChartMap = LanguageMapToChartMap;
exports.ThemeChartMap = ThemeChartMap;
exports.bollKList = bollKList;
exports.kdjKList = kdjKList;
exports.macdKList = macdKList;
exports.shareTimeList = shareTimeList;
exports.timeLocaleLanguageMap = timeLocaleLanguageMap;
exports.timeMap = timeMap;
exports.tradingviewTimeMap = tradingviewTimeMap;
